src/orientation.h

```diff
@@ -95,4 +95,44 @@ static inline bool IsBackEdge(ViewOrientation orient, TileEdge edge)
 XYZPoint16 OrientatedOffset(const uint8 orientation, const int x, const int y);
 XYZPoint16 UnorientatedOffset(const uint8 orientation, const int x, const int y);
 
+/** Information about the graphics sizes at a zoom scale. */
+struct ZoomScale {
+	int tile_width;   ///< Width  of a tile in pixels.
+	int tile_height;  ///< Height of a tile in pixels.
+};
+
+/** Available zoom scales, sorted from smallest to biggest. */
+constexpr ZoomScale _zoom_scales[] = {
+	{ 16,  4},
+	{ 32,  8},
+	{ 64, 16},
+	{128, 32},
+	{256, 64},
+};
```

Comment:
Not sure it's useful to hard-code these.

The factor 4 between the first and second number is from the isometric view, so you never get rid of that. Then the first value is a simple power of 2, although with your zoom idea maybe it's possible to have any width as long as it's a multiple of 4.

But for now power of 2 is enough and probably simplest to implement. It's just that at larger sizes doubling the size is a big step.

Reply:
I changed it to add more scales (heights 24 and 48) and automatic width calculation.